core/src/main/java/uk/gov/justice/services/core/enveloper/spi/DefaultEnveloperProvider.java

```diff
@@ -0,0 +1,22 @@
+package uk.gov.justice.services.core.enveloper.spi;
+
+import uk.gov.justice.services.core.enveloper.DefaultEnveloper;
+import uk.gov.justice.services.core.enveloper.Enveloper;
+import uk.gov.justice.services.messaging.Envelope;
+import uk.gov.justice.services.messaging.JsonEnvelope;
+
+import java.util.function.Function;
+
+public class DefaultEnveloperProvider implements EnveloperProvider {
```

Comment:
This class needs to bridge the gap between the static, SPI world and the CDI managed bean world. The DefaultEnveloper should be a managed bean, and this non-managed class should get a reference to that bean and use it to call non-static methods. Currently, you are using static field in the DefaultEnveloper to trick CDI into injecting beans into its instance of the DefaultEnveleper and using the static fields to pass those beans to the instance that SPI has created, which is all a bit of a nasty workaround.